Rakefile

```diff
@@ -121,6 +122,21 @@ task apply: [:configure_state] do
 end
 
 
+desc 'Latest Lambda version ID'
+task :latest_version_id do
+lambda_filename = ENV['TF_VAR_LAMBDA_FILENAME']
+  if lambda_filename
+    s3 = Aws::S3::Resource.new(region: 'eu-west-1')
+    bucket = s3.bucket("govuk-lambda-applications-#{deploy_env}")
+    bucket.object_versions.each do |obj|
+     if obj.key == lambda_filename && obj.is_latest
+       ENV['TF_VAR_LAMBDA_VERSIONID'] = obj.version_id
+     end
+   end
```

Comment:
Instead of iterating over all of the `bucket.object_versions` (which might grow quite quickly) I think calling `bucket.object(lambda_filename).version_id` will return the latest version. It might also be worth guarding against the case where that value is `"null"` (which I think is what happens when versioning isn't enabled).